autoload/ale/completion.vim

```diff
@@ -183,52 +185,71 @@ function! ale#completion#RestoreCompletionOptions() abort
     endif
 endfunction
 
+function! ale#completion#FindCompletionStart() abort
+    let [l:line, l:column] = getcurpos()[1:2]
+    let l:regex = s:GetFiletypeValue(s:omni_start_map, &filetype)
+    let l:up_to_column = getline(l:line)[: l:column - 2]
+    let l:match = matchstr(l:up_to_column, l:regex)
+
+    let b:ale_completion_start = l:column - len(l:match) - 1
+
+    return b:ale_completion_start
+endfunction
+
+function! ale#completion#PollCompletionResults() abort
+    if ale#completion#Queue()
+        return []
+    else
+        return get(b:, 'ale_completion_result', [])
+    endif
+endfunction
+
```

Comment:
I think you can pretty much undo the changes here. The code is basically the same, and the changes don't really improve on what was there. The previous behavior of looking at the line and column as requested instead of the current cursor position may be more appropriate.

Reply:
Is this referring to the whole block of changes above? `ale#completion#PollCompletionResults `and `ale#completion#FindCompletionStart`

Those were exactly what was before inside of ale#completion#OmniFunc, only turned into functions because external sources don't use the flow of omnifunctions to command their behavior.

Reply:
Cool, thank you for clarifying. That makes sense.

Remember to add tests for these function names.

Let's add the deoplete source file to ALE's repo itself.